src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -568,14 +568,29 @@ void MulticopterPositionControl::Run()
 
 			_control.setState(states);
 
+			const hrt_abstime now = hrt_absolute_time();
+
 			// Run position control
-			if (!_control.update(dt)) {
-				// Failsafe
-				_vehicle_constraints = {0, NAN, NAN, false, {}}; // reset constraints
+			if (_control.update(dt)) {
+
+				// Valid control update - store for fallback
+				_last_valid_setpoint_time = now;
```

Comment:
The `trajectory_setpoint_s` has a timestamp field. I'd check a timeout against that.